ts/store/reducers/wallet.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * A reducer for the wallet, with some
+ * selectors and type guards
+ */
+
+import { Action } from "../../actions/types";
+import { UNKNOWN_TRANSACTION, WalletTransaction } from "../../types/wallet";
+import {
+  LOAD_LATEST_TRANSACTIONS_LIST,
+  LOAD_TRANSACTIONS_LIST_BY_SELECTED_CARD,
+  SELECT_CARD,
+  SHOW_TRANSACTION_DETAILS,
+  TRANSACTIONS_FETCHED
+} from "../actions/constants";
+
+export type EmptyState = Readonly<{
+  hasTransactionForDetails: boolean;
```

Comment:
This definition has gotten out of hand and needs to be redefined so as to limit `EmptyState` to -- well -- an empty state